test/integration/sagemaker/test_default_inference.py

```diff
@@ -0,0 +1,114 @@
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import json
+import numpy as np
+import pytest
+import requests
+import sagemaker
+from sagemaker.predictor import RealTimePredictor
+from sagemaker.pytorch import PyTorchModel, PyTorchPredictor
+
+from integration import (
+    default_model_script,
+    default_model_tar,
+    default_model_eia_script,
+    default_model_eia_tar,
+)
+from integration.sagemaker.timeout import timeout_and_delete_endpoint
+
+
+@pytest.mark.cpu_test
+def test_default_inference_cpu(sagemaker_session, image_uri, instance_type):
+    instance_type = instance_type or "ml.c4.xlarge"
+    # Scripted model is serialized with torch.jit.save().
+    # Default inference test doesn't need to instantiate model definition
+    _test_default_inference(sagemaker_session, image_uri, instance_type, default_model_tar, default_model_script)
+
+
+@pytest.mark.gpu_test
+def test_default_inference_gpu(sagemaker_session, image_uri, instance_type):
+    instance_type = instance_type or "ml.p2.xlarge"
+    # Scripted model is serialized with torch.jit.save().
+    # Default inference test doesn't need to instantiate model definition
+    _test_default_inference(sagemaker_session, image_uri, instance_type, default_model_tar, default_model_script)
+
+
+@pytest.mark.skip(reason="Latest EIA version - 1.5.1 uses mms. Enable when EIA images use torchserve")
+@pytest.mark.eia_test
+def test_default_inference_eia(sagemaker_session, image_uri, instance_type, accelerator_type):
+    instance_type = instance_type or "ml.c4.xlarge"
+    # Scripted model is serialized with torch.jit.save().
+    # Default inference test doesn't need to instantiate model definition
+    _test_default_inference(
+        sagemaker_session,
+        image_uri,
+        instance_type,
+        default_model_eia_tar,
+        default_model_eia_script,
+        accelerator_type=accelerator_type,
+    )
+
+
+def _test_default_inference(
+    sagemaker_session, image_uri, instance_type, model_tar, mnist_script, accelerator_type=None
+):
+    endpoint_name = sagemaker.utils.unique_name_from_base("sagemaker-pytorch-serving")
+
+    model_data = sagemaker_session.upload_data(
+        path=model_tar,
+        key_prefix="sagemaker-pytorch-serving/models",
+    )
+
+    pytorch = PyTorchModel(
+        model_data=model_data,
+        role="SageMakerRole",
+        predictor_cls=RealTimePredictor if not accelerator_type else PyTorchPredictor,
+        entry_point=mnist_script,
+        image=image_uri,
+        sagemaker_session=sagemaker_session,
+    )
+    with timeout_and_delete_endpoint(endpoint_name, sagemaker_session, minutes=30):
+        # Use accelerator type to differentiate EI vs. CPU and GPU. Don't use processor value
+        if accelerator_type is not None:
```

Comment:
Isn't this branch pointless? If it is None, then the code will do the same thing but use the default value of `accelerator_type` parameter, which is most likely `None` (though I didn't check)

Reply:
You are right. The default for `accelerator_type` is None. So the if and else conditions can both be collapsed into one single implementation.